### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by DaveZag**
# OOP school library: set up associations
### On this branch I:
- Add 3 more classes(book, classroom, and rental).
- Add a many-to-many association between book and person using the rental class.
- Add a many-to-one association between student and classroom.
- Fix linters errors.

### CODE REVIEW

File: modules/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (franclobo) [inline modules/classroom.rb]: - [x] Kindly make sure that the relationship between classroom and student is created i.e. when adding a student to a classroom it also sets the classroom for the student 👉🏽 `attr_reader :students`

File: modules/student.rb [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+  attr_reader :classroom
+
   def initialize(classroom, age, name = 'unknown', parent_permission: true)
     super(age, name, parent_permission)
     @classroom = classroom
   end
 
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.students.push(self) unless classroom.students.includes?(self)
```
Comment (franclobo) [inline modules/student.rb]: - [x] Please 🙏🏽 correct the method `include` instead of `includes`.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @DaveZag, 

# ♻️ REQUIRED CHANGES ♻️ 

soClose

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- Professional README file ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/student.rb (hunk 2 above)
Comment (DaveZag) [reply modules/student.rb]: Thanks for the feedback. All the requested changes have been implemented.

**Review by codecaiine — APPROVED** [review]
Hi @DaveZag , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great project implementation :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**